chajda/tsquery/augments.py

```diff
@@ -52,62 +52,185 @@ def augments_gensim(lang, word, config=Config(), n=5):
+    >>> augments_fasttext('en','weapon', n=5, annoy=False)
     ['weaponthe', 'weopon']
 
-    >>> augments_fasttext('en','king', n=5)
+    >>> augments_fasttext('en','king', n=5, annoy=False)
     ['queen', 'kingthe']
+    
+    NOTE:
+    Populating an AnnoyIndex with vectors from a fasttext model requires more memory than github actions allows for.
+    Therefore, the doctests involving the annoy library have been commented out below.
+
+    # >>> augments_fasttext('en','weapon', n=5)
+    # ['nonweapon', 'loadout', 'dualwield', 'autogun']
+
+    # >>> augments_fasttext('en','king', n=5)
+    # ['kingthe', 'kingly']
 
     NOTE:
     Due to the size of fasttext models (>1gb),
     testing multiple languages in the doctest requires more space than github actions allows for.
     For this reason, tests involving languages other than English have been commented out below.
 
-    #>>> augments_fasttext('ja','さようなら', n=5)
-    #['さよなら', 'バイバイ', 'サヨウナラ', 'さらば', 'おしまい']
+    # >>> augments_fasttext('ja','さようなら', n=5)
+    # ['さよなら', 'バイバイ', 'サヨウナラ', 'さらば', 'おしまい']
 
-    #>>> augments_fasttext('es','escuela', n=5)
-    #['escuelala', 'academia', 'universidad', 'laescuela']
+    # >>> augments_fasttext('es','escuela', n=5)
+    # ['escuelala', 'academia', 'universidad', 'laescuela']
     '''
-
     # download and load the fasttext model if it's not already loaded
     try:
         fasttext_models[lang]
     except KeyError:
-        with suppress_stdout_stderr():
-            fasttext.util.download_model(lang, if_exists='ignore')
-        fasttext_models[lang] = fasttext.load_model('cc.{0}.300.bin'.format(lang))
+        load_fasttext_model(lang)
+    
+    # augments_fasttext defaults to using the annoy library to find nearest neighbors,
+    # if annoy==False is passed into the function, then the fasttext library will be used.
+    if annoy:
+
+        # populate and load AnnoyIndex with vectors from fasttext model if not already populated
+        create_annoy_index_if_needed(lang)
 
-    # find the most similar words
-    topn = fasttext_models[lang].get_nearest_neighbors(word, k=n)
-    words = ' '.join([ word for (rank, word) in topn ])
+        # find the most similar words using annoy library
+        n_nearest_neighbor_indices = annoy_indices[lang].get_nns_by_vector(fasttext_models[lang][word], n)
+        n_nearest_neighbors = [fasttext_models[lang].words[i] for i in n_nearest_neighbor_indices]
+        words = ' '.join([ word for word in n_nearest_neighbors ])
+
+    else:
+        # find the most similar words using fasttext library
+        topn = fasttext_models[lang].get_nearest_neighbors(word, k=n)
+        words = ' '.join([ word for (rank, word) in topn ])
 
     # lemmatize the results so that they'll be in the search document's vocabulary
     words = lemmatize(lang, words, add_positions=False, config=config).split()
     words = list(filter(lambda w: len(w)>1 and w != word, words))[:n]
 
+    
     return words
 
+
```

Comment:
this second newline actually brings the code into conformance with PEP8, so is in principle a good addition, but we still don't want to add it to this commit; the reason is that we want every commit to be just a single "idea" so that it's easy to understand exactly what's going on in the commit when looking at it.  This is important both for simplifying code review, but also is crucial for debugging large projects where you have to find exactly which commit introduced a certain bug and revert the commit; (once you find the offending commit, you don't want to have to revert a totally separate feature just because it was included in the same commit)

in general, it's a good practice to get into to force yourself to only do one thing at a time in the code, commit that one thing, then move on to something else